exp/controllers/rosamachinepool_controller_test.go

```diff
@@ -60,3 +81,459 @@ func TestNodePoolToRosaMachinePoolSpec(t *testing.T) {
+
+			r := ROSAMachinePoolReconciler{
+				Recorder:         recorder,
+				WatchFilterValue: "",
+				Endpoints:        []scope.ServiceEndpoint{},
+				Client:           testEnv,
+				NewStsClient:     func(cloud.ScopeUsage, cloud.Session, logger.Wrapper, runtime.Object) stsiface.STSAPI { return stsMock },
+				NewOCMClient: func(ctx context.Context, rosaScope *scope.ROSAControlPlaneScope) (rosa.OCMClient, error) {
+					return ocmMock, nil
+				},
+			}
+
+			req := ctrl.Request{}
+			req.NamespacedName = types.NamespacedName{Name: test.old.Name, Namespace: ns.Name}
+
+			result, errReconcile := r.Reconcile(ctx, req)
+			g.Expect(errReconcile).ToNot(HaveOccurred())
+			g.Expect(result).To(Equal(test.result))
+			time.Sleep(50 * time.Millisecond)
+
+			m := &expinfrav1.ROSAMachinePool{}
+			key := client.ObjectKey{Name: test.old.Name, Namespace: ns.Name}
+			errGet := testEnv.Get(ctx, key, m)
+			g.Expect(errGet).NotTo(HaveOccurred())
+			g.Expect(m.Status.Ready).To(Equal(test.new.Status.Ready))
+			g.Expect(m.Status.Replicas).To(Equal(test.new.Status.Replicas))
+			g.Expect(m.Status.ID).To(Equal(test.new.Status.ID))
+
+			// cleanup
+			for _, obj := range objects {
+				cleanupObject(g, obj)
+			}
+			mockCtrl.Finish()
+		})
+	}
+
+	t.Run("Reconcile delete", func(t *testing.T) {
+		g := NewWithT(t)
+		mockCtrl := gomock.NewController(t)
+		recorder := record.NewFakeRecorder(10)
+		ctx := context.TODO()
+		mp := &expinfrav1.ROSAMachinePool{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:       "rosa-machinepool-9",
+				Namespace:  ns.Name,
+				UID:        "rosa-machinepool-9",
+				Finalizers: []string{expinfrav1.RosaMachinePoolFinalizer},
+			},
+			TypeMeta: metav1.TypeMeta{
+				Kind:       "ROSAMachinePool",
+				APIVersion: expinfrav1.GroupVersion.String(),
+			},
+			Spec: expinfrav1.RosaMachinePoolSpec{
+				NodePoolName: "test-nodepool-1",
+				Version:      "4.14.5",
+
+				Subnet:       "subnet-id",
+				InstanceType: "m5.large",
+			},
+		}
+		oc := ownerCluster(9)
+		omp := ownerMachinePool(9)
+		cp := rosaControlPlane(9)
+		objects := []client.Object{oc, omp, cp, mp}
+
+		for _, obj := range objects {
+			createObject(g, obj, ns.Name)
+		}
+
+		cpPh, err := patch.NewHelper(cp, testEnv)
+		cp.Status.Ready = true
+		cp.Status.ID = "rosa-control-plane-9"
+		g.Expect(cpPh.Patch(ctx, cp)).To(Succeed())
+		g.Expect(err).ShouldNot(HaveOccurred())
+
+		ocmMock := mocks.NewMockOCMClient(mockCtrl)
+		expect := func(m *mocks.MockOCMClientMockRecorder) {
+			m.GetNodePool(gomock.Any(), gomock.Any()).DoAndReturn(func(clusterId string, nodePoolID string) (*cmv1.NodePool, bool, error) {
+				nodePoolBuilder := nodePoolBuilder(mp.Spec, omp.Spec)
+				nodePool, err := nodePoolBuilder.ID("node-pool-1").Build()
+				g.Expect(err).NotTo(HaveOccurred())
+				return nodePool, true, nil
+			}).Times(1)
+			m.DeleteNodePool("rosa-control-plane-9", "node-pool-1").DoAndReturn(func(clusterId string, nodePoolID string) error {
```

Comment:
where is this name `"rosa-control-plane-9"` come from ? if it is the cp CR then better is `cp.Name`